src/main/kotlin/ru/kode/detekt/rule/compose/UnnecessaryLayoutWrapper.kt

```diff
@@ -0,0 +1,75 @@
+package ru.kode.detekt.rule.compose
+
+import io.gitlab.arturbosch.detekt.api.CodeSmell
+import io.gitlab.arturbosch.detekt.api.Config
+import io.gitlab.arturbosch.detekt.api.Debt
+import io.gitlab.arturbosch.detekt.api.Entity
+import io.gitlab.arturbosch.detekt.api.Issue
+import io.gitlab.arturbosch.detekt.api.Rule
+import io.gitlab.arturbosch.detekt.api.Severity
+import org.jetbrains.kotlin.psi.KtBlockExpression
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtLambdaArgument
+import org.jetbrains.kotlin.psi.KtNameReferenceExpression
+import org.jetbrains.kotlin.psi.psiUtil.getChildOfType
+import org.jetbrains.kotlin.psi.psiUtil.getChildrenOfType
+
+class UnnecessaryLayoutWrapper(config: Config = Config.empty) : Rule(config) {
+
+  override val issue: Issue = Issue(
+    javaClass.simpleName,
+    Severity.CodeSmell,
+    "Reports when Box, Column or Row is used as an unnecessary wrapper",
+    Debt.FIVE_MINS
+  )
+
+  private val layoutNames = setOf("Box", "Column", "Row")
+
+  private val unnecessaryExpressions = mutableListOf<KtCallExpression>()
```

Comment:
There's no need to accumulate expressions and report them in `postVisit`, you can report them right when you find them, i.e. instead of having

```kotlin
unnecessaryExpressions.add(expression)
```
just call 
```
report(...)
```

Detekt will visit all call expressions anyway, so no need to complicate with `preVisit`, `postVisit` and state here...